closure/goog/array/array.js

```diff
@@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
  *     for every element. This function takes 3 arguments (the element, the
  *     index and the array) and should return a boolean.
  * @param {S=} opt_obj An optional "this" context for the function.
- * @return {T|null} The first array element that passes the test, or null if no
- *     element is found.
+ * @return {T|string|null} The first array element that passes the test, or
```

Comment:
I'm really not excited to have string implement IArrayLike.   It seems more
likely to be an error than correct code.

On Fri, Feb 12, 2016 at 4:05 PM, Michael Zhou <notifications@github.com>
wrote:

> In closure/goog/array/array.js
> <https://github.com/google/closure-library/pull/655#discussion_r52815312>:
>
> > @@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
> >   *     for every element. This function takes 3 arguments (the element, the
> >   *     index and the array) and should return a boolean.
> >   * @param {S=} opt_obj An optional "this" context for the function.
> > - * @return {T|null} The first array element that passes the test, or null if no
> > - *     element is found.
> > + * @return {T|string|null} The first array element that passes the test, or
>
> Making string implement IArrayLike seems better. Would it be hard to land
> such a fix internally though? @concavelenz
> <https://github.com/concavelenz>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/closure-library/pull/655/files#r52815312>.
>